### PR TITLE
[redux-form] - Fix test and linting errors

### PR DESCRIPTION
**Pull request by tehbi4**
Fix failing test (#20414) and linting errors.

Please fill in this template.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: `SubmitHandler` type (in `types/redux-form/lib/reduxForm.d.ts`) was totally changed in accordance with [redux-form flow declaration](https://github.com/erikras/redux-form/blob/84693b5a496fc36fe89ac0603b84084d028d5bb5/src/handleSubmit.js#L11)
- [ ] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by tehbi4** [comment]
@CarsonF  @aikoven @LKay @bancek @alsiola
For some reason @dt-bot ignored this PR
Can you approve it?

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Comment by RyanCavanaugh** [comment]
@tehbi4 Please address the merge conflict.

**Comment by shepmaster** [comment]
poke — there's still a merge conflict

File: types/redux-form/lib/reduxForm.d.ts [hunk 1]
```diff
@@ -15,15 +15,18 @@ import {
...
+} from "../index";
 
 export type FormSubmitHandler<FormData = {}, P = {}> =
     (values: Partial<FormData>, dispatch: Dispatch<any>, props: P) => void | FormErrors<FormData> | Promise<any>;
 
-export interface SubmitHandler<FormData = {}, P = {}> {
-    (handler: FormSubmitHandler<FormData, P>): FormEventHandler<any>;
-    (event: SyntheticEvent<any>): void;
-}
+export type SubmitHandler<FormData = {}, P = {}> = (
```
Comment (19majkel94) [inline types/redux-form/lib/reduxForm.d.ts]: Why this lane has been changed? This makes my code doesn't compile at all 😠 
image

Comment (mhegazy) [reply types/redux-form/lib/reduxForm.d.ts]: @19majkel94 feel free to send a new PR and we will merge it in

Comment (tehbi4) [reply types/redux-form/lib/reduxForm.d.ts]: @19majkel94 Hi! Fixed in #22945, could you please review it?

**Pull request merged**